app/lib/package/tarball_storage.dart

```diff
@@ -16,13 +17,14 @@ import 'models.dart';
 final _logger = Logger('package_storage');
 
 /// The GCS object name of a tarball object - excluding leading '/'.
+@visibleForTesting
 String tarballObjectName(String package, String version) =>
     '${_tarballObjectNamePackagePrefix(package)}$version.tar.gz';
 
 /// The GCS object prefix of a tarball object for a given [package] - excluding leading '/'.
 String _tarballObjectNamePackagePrefix(String package) => 'packages/$package-';
 
-class PackageStorage {
+class TarballStorage {
```

Comment:
Is our thinking that this object will mainly be responsible for the `canonicalBucket`, right?

As eventually, the public bucket will go out of use, when `ApiExporter` starts updating `ExportedApi` and we've migrated the load balancer to use this.

Note. I don't mind that we let this object keep responsibility for the public bucket, until the public bucket is phased out.

Reply:
Yes, correct.